src/types/mock.rs

```diff
@@ -0,0 +1,98 @@
+#![allow(unknown_lints)]
+#![warn(clippy::all)]
+
+use crate::protocol_traits::graph::{Graph, GraphObject};
+use crate::types::network::{Artifact, DependencyType, Network};
+use crate::util::quickcheck::frequency;
+use quickcheck::{Arbitrary, Gen};
+use rand::Rng;
+
+pub type MockNetwork = Network<f64>;
+
+#[derive(Debug)]
+struct ArbitraryEdge<'a> {
+    source: &'a String,
+    target: &'a String,
+    id: usize,
+    metadata: DependencyType<f64>,
+}
+
+impl Arbitrary for MockNetwork {
+    // Tries to generate an arbitrary Network.
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        let mut graph = Network::default();
+        let nodes: Vec<Artifact<String>> = Arbitrary::arbitrary(g);
+
+        let sub_nodes = nodes
+            .iter()
+            .cloned()
+            .take(3)
```

Comment:
Does this mean the graph will only have 3 nodes?

Reply:
Ignore, brainfart (see below) 😬